R/form_schema_ext.R

```diff
@@ -264,72 +243,114 @@ form_schema_ext <- function(flatten = FALSE, odata = FALSE, parse = TRUE, pid =
 
       ### PART 1.1: parse choice labels
       ## checks existence of  choice list:
-      choice_items <- xml2::xml_find_all(xml2::xml_parent(this_rawlabel), "./item")
+      choice_items <- xml2::xml_find_all(
+        xml2::xml_parent(this_rawlabel), "./item"
+      )
+
       if (length(choice_items) > 0) {
 
         # check if 'choices' column already exist
         if (!("choices" %in% colnames(extension))) {
 
           # if not, create new column
-          extension <- cbind(extension, data.frame(choices = rep(NA, nrow(extension))))
+          extension <- cbind(extension, data.frame(
+            choices = rep(NA, nrow(extension))
+          ))
         }
 
         # initialize lists
         choice_values <- list()
         choice_labels <- list()
 
         # iterate through choice list:
-        for (jj in 1:length(choice_items)) {
+        for (jj in seq_along(choice_items)) {
 
           ## reads choice item
           this_choiceitem <- choice_items[jj]
 
           # value
-          this_choicevalue <- xml2::xml_text(xml2::xml_find_first(this_choiceitem, "./value"))
+          this_choicevalue <- xml2::xml_text(
+            xml2::xml_find_first(this_choiceitem, "./value")
+          )
           choice_values[jj] <- this_choicevalue
 
           # raw label
-          this_rawchoicelabel <- xml2::xml_find_first(this_choiceitem, "./label")
+          this_rawchoicelabel <- xml2::xml_find_first(
+            this_choiceitem, "./label"
+          )
 
           # first checks if choice label is mapped with a translation function
-          has_translation_choice <- xml2::xml_has_attr(this_rawchoicelabel, "ref")
+          has_translation_choice <- xml2::xml_has_attr(
+            this_rawchoicelabel, "ref"
+          )
 
           if (has_translation_choice) {
-            id_choice <- sub("')", "", sub("jr:itext\\('", "", xml2::xml_attr(this_rawchoicelabel, "ref")))
-            choice_translations <- all_translations[xml2::xml_attr(all_translations, "id") == id_choice]
+            id_choice <- sub(
+              "')",
+              "",
+              sub(
+                "jr:itext\\('",
+                "",
+                xml2::xml_attr(this_rawchoicelabel, "ref")
+              )
+            )
+
+            choice_translations <- all_translations[xml2::xml_attr(
+              all_translations, "id"
+            ) == id_choice]
 
 
             # iterate through choice translations
-            for (kk in 1:length(choice_translations)) {
+            for (kk in seq_along(choice_translations)) {
```

Comment:
Here's a chance to factor out a function and purrr::map() or pmap() over the data, but the code as is with the inline comments reads nicely enough. I'd be happy to keep as is. No need to go overboard with functional pur(rr)itanism here.

Reply:
I indeed prefer the long way.. it is easier to understand and follow